src/libs/Network/MainQueue.js

```diff
@@ -56,17 +55,14 @@ function process() {
     // Some requests should be retried and will end up here if the following conditions are met:
     // - we are in the process of authenticating and the request is retryable (most are)
     // - the request does not have forceNetworkRequest === true (this will trigger it to process immediately)
-    // - the request does not have shouldRetry === false (specified when we do not want to retry, defaults to true)
     const requestsToProcessOnNextRun = [];
 
     _.each(networkRequestQueue, (queuedRequest) => {
         // Check if we can make this request at all and if we can't see if we should save it for the next run or chuck it into the ether
         if (!canMakeRequest(queuedRequest)) {
-            const shouldRetry = lodashGet(queuedRequest, 'data.shouldRetry');
-            if (shouldRetry) {
+            // If we can't make the request because the sequential queue is running, let's wait until it finished then run the request instead of tossing it
+            if (SequentialQueue.isRunning) {
```

Comment:
If we can't make the request right now I think we should push it to process on the next run in all cases. The only time we skipped it before is if we explicitly set `shouldRetry: false` and we no longer do that.

Reply:
Hmm I'm not sure I agree. The other time it would fail is if we're authenticating, which could fail and we wouldn't want to retry no?

Reply:
I don't think this block is really about "retrying". If a request can be made, it's not added to requestsToProcessOnNextRun. If it can't be made then we should add it to requestsToProcessOnNextRun so that we can do the request in the future.

I'm not seeing how we could ever end up retrying the same request.

Reply:
So if a request from the main queue arrives here, and we fail to re-authenticate. Why would we queue this to run later? 
If we fail to re-authenticate we're logging the user out and we're clearing all the requests anyways no?

Reply:
> I don't think this block is really about "retrying". If a request can be made, it's not added to requestsToProcessOnNextRun

+1. I am also not sure what we're doing in this block is correct.
Should we check `isFromSequentialQueue` and then retry?

Reply:
@MonilBhavsar this is only MainQueue requests, so `isFromSequentialQueue` can never be true. 

@neil-marcellini I'm fine just requeueing everything, but just pointing this out again.

> So if a request from the main queue arrives here, and we fail to re-authenticate. Why would we queue this to run later?
If we fail to re-authenticate we're logging the user out and we're clearing all the requests anyways no?